crates/nix_rs/src/env.rs

```diff
@@ -196,6 +202,86 @@ impl OS {
     }
 }
 
+/// The installer used to install Nix (applicable only for non-NixOS systems)
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum NixInstaller {
+    /// The installer from <https://github.com/DeterminateSystems/nix-installer>
+    DetSys { version: NixInstallerVersion },
+    /// Either offical installer or from a different package manager
+    Other,
+}
+
+impl Display for NixInstaller {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            NixInstaller::DetSys { version } => {
+                write!(f, "nix-installer ({})", version)
+            }
+            NixInstaller::Other => write!(f, "Unknown installer"),
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, PartialOrd, Eq, Clone)]
+pub struct NixInstallerVersion {
+    pub major: u32,
+    pub minor: u32,
+    pub patch: u32,
+}
+
+impl Display for NixInstallerVersion {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}.{}.{}", self.major, self.minor, self.patch)
+    }
+}
+
+#[derive(Error, Debug, Clone, PartialEq)]
+pub enum BadInstallerVersion {
+    #[error("Regex error: {0}")]
+    Regex(#[from] regex::Error),
+    #[error("Failed to decode installer output: {0}")]
+    Decode(#[from] std::string::FromUtf8Error),
+    #[error("Failed to parse installer version: {0}")]
+    Parse(#[from] std::num::ParseIntError),
+    #[error("Failed to fetch installer version")]
+    Command,
+}
+
+impl NixInstallerVersion {
+    pub fn get_version(executable_path: &Path) -> Result<Self, BadInstallerVersion> {
+        let output = std::process::Command::new(executable_path)
+            .arg("--version")
+            .output()
+            .map_err(|_| BadInstallerVersion::Command)?;
```

Comment:
> `_`

We should not ignore the `Command` error here.

Reply:
https://github.com/juspay/omnix/pull/213/commits/8e9a10d667143f75b74366c7fc5c792e488fd710